src/asciidoc/reference/packages/withkojiDispatch.adoc

```diff
@@ -51,19 +48,177 @@ const dispatch = new Dispatch({
 });
 ----
 
+== Dispatch properties
+
+[.hcode, id=".clientId", reftext="clientId"]
+=== .clientId
+
+_String_, ID of the currently connected client.
+
+[.hcode, id=".connectedClients", reftext="connectedClients"]
+=== .connectedClients
+
+_Object_, contains IDs of currently connected clients as keys and an object of their user data.
+
+==== Example
+[source,json]
+----
+{
+	"0dc15e36-81b4-442b-be4d-bf590ee73c2a": {
+		username: "test1",
+		lastPing: 1600218622595
+	},
+	"1fe2eb80-57e3-435f-9314-faa74853c34a": {
+		username: "test2",
+		lastPing: 1600218623097
+	}
+}
+----
+
 [.hcode, id=".latency", reftext="latency"]
-=== .latency()
+=== .latency
+
+_Number_, the latency of the last message in milliseconds.
+
+[.hcode, id=".shardName", reftext="shardName"]
+=== .shardName
+
+_String_, name of the shard the client is currently connected to.
+
+[.hcode, id=".userInfo", reftext="userInfo"]
+=== .userInfo
+
+_Object_, the user info of the currently connected client.
+
+== Dispatch methods
+
+[.hcode, id=".connect", reftext="connect"]
+=== .connect()
+
+(Async) Connects the dispatch client to an open shard.
+
+==== Example
+
+[source,javascript]
+dispatch.connect();
 
-Checks the latency of dispatch messages.
+[.hcode, id=".disconnect", reftext="disconnect"]
+=== .disconnect()
+
+(Async) Disconnects the dispatch client from the active connection. 
+
+==== Example
+
+[source,javascript]
+dispatch.disconnect();
+
+[.hcode, id=".emitEvent", reftext="emitEvent"]
+=== .emitEvent(eventName, payload)
+
+Emits the named event to all clients on the current shard.
+
+==== Parameters
+
+* `eventName` – _String_, name of the event.
+* `payload` – _Any_, data to send with event.
+
+==== Example
+
+[source,javascript]
+dispatch.emitEvent('myEvent', myDataPayload);
+
+[.hcode, id=".info", reftext="info"]
+=== .info()
+
+Gets info on the shard the client is currently connected to.
 
 ==== Returns
 
-Number, the latency of the last message in milliseconds.
+(Async) _ShardInfo_, Object describing the shard the client is currently connected to with shard name and number of connected clients.
+
+==== Example
+
+[source,javascript]
+----
+dispatch.info.then((info) => {
+    currentInfo = info;
+});
+----
+
+[.hcode, id=".on", reftext="on"]
+=== .on(eventName, handler)
+
+Sets a listener to a specific event and runs the handler when the event is dispatched over the shard.
+
+TIP: For Koji custom events see <<#_dispatch_event, DISPATCH_EVENT>>.
```

Comment:
I don't think this belongs as a tip because it seems like required information to use the method. Maybe add it to the `eventName` description? E.g., "For the available events, see ..."

Also, could this method also be applied for custom events via `emitEvent`? Maybe a cross-ref?

Reply:
Latest commits adds in a comment regarding emitEvent also, but perhaps you are correct in saying it is required info. Will move out of TIP.